pkg/controller/core.oam.dev/v1alpha2/application/apply.go

```diff
@@ -494,6 +508,9 @@ func (h *appHandler) generateResourceTrackerName() string {
 // with application
 func (h *appHandler) checkCrossNamespace(u *unstructured.Unstructured) (bool, error) {
 	gk := u.GetObjectKind().GroupVersionKind().GroupKind()
+	if gk == types.AutoDetectWorkloadDefinitionGVK.GroupKind() {
```

Comment:
This is workable. But it's better to avoid invading `checkCXNamespace`'s logic with "check auto-detect" logic, these two logic are not closely relevant, even though `checkCXNamespace` is the place where an error occurs.  How about moving this check into an independent function?

Reply:
good suggestion, it's better to implement an independent function